src/Elcodi/Bundle/SitemapBundle/DependencyInjection/ElcodiSitemapExtension.php

```diff
@@ -113,13 +110,16 @@ protected function loadBlocks(array $config, ContainerBuilder $container)
         foreach ($blocks as $blockName => $block) {
             $container
                 ->register(
-                    'elcodi.sitemap_entity_loader.' . $blockName,
-                    '%elcodi.core.sitemap.loader.entity_loader.class%'
+                    'elcodi.sitemap_element_generator.entity_' . $blockName,
+                    'Elcodi\Component\Sitemap\Element\EntitySitemapElementGenerator'
                 )
+                ->addArgument(new Reference('elcodi.factory.sitemap_element'))
```

Comment:
Take in account that this code is computed in cache-warm-up time, so it is not really important at all... Both ways are right indeed, so if you want to purpose a full change of this strategy (with pros and contras), then you can expose your POV in a RFC issue :)